addons/website_slides/models/slide_channel.py

```diff
@@ -680,68 +781,129 @@ def action_redirect_to_members(self, completed=False):
+        return invited_res or bool((self - courses_invited)._action_add_members(self.env.user.partner_id))
+
+    def _action_add_members(self, target_partners, member_status='joined'):
+        """ Adds the target_partners as attendees of the channel(s).
+            Partners are added as follows, depending on the value of member_status:
+            1) (Default) 'joined'. The partners will be added as enrolled attendees. This will make the content
+                (slides) of the channel available to that partner. This can also happen when an invited attendee
+                enrolls themself. The attendes are also subscribed to the chatter of the channel.
+            2) 'invited' : This is used when inviting partners. The partners are added as invited attendees
+                This will make the channel accessible but not the slides until they enroll themselves.
+
+        Returns the union of previous partners renrolling, new partners and invited ones enrolling themselves."""
+        # Channels on which the current user has the rights to write.
+        channels_to_join = self._filter_add_members(target_partners)
+        SlideChannelPartnerSudo = self.env['slide.channel.partner'].sudo()
+        if channels_to_join:
+
+            # Existing channel partners
             existing_channel_partners = self.env['slide.channel.partner'].with_context(active_test=False).sudo().search([
-                ('channel_id', 'in', self.ids),
+                ('channel_id', 'in', channels_to_join.ids),
                 ('partner_id', 'in', target_partners.ids)
             ])
-            existing_map = defaultdict(list)
-            for item in existing_channel_partners:
-                existing_map[item.channel_id.id].append(item.partner_id.id)
-
-            archived_members = existing_channel_partners.filtered(lambda channel_partner: not channel_partner.active)
-            if archived_members:
-                archived_members.action_unarchive()
-
-            to_create_values = [
-                dict(channel_id=channel.id, partner_id=partner.id)
-                for channel in to_join
-                for partner in target_partners if partner.id not in existing_map[channel.id]
-            ]
-            slide_partners_sudo = self.env['slide.channel.partner'].sudo().create(to_create_values)
-            to_join.message_subscribe(partner_ids=target_partners.ids, subtype_ids=[self.env.ref('website_slides.mt_channel_slide_published').id])
-            # Also consider re-joined members along with newly joined ones
-            return slide_partners_sudo | archived_members
-        return self.env['slide.channel.partner'].sudo()
+
+            # Unarchive existing channel partners
+            archived_channel_partners = existing_channel_partners.filtered(lambda channel_partner: not channel_partner.active)
+            unarchived_channel_partners_sudo = SlideChannelPartnerSudo
+            if member_status != 'invited' and archived_channel_partners:
```

Comment:
Would use "member_status == 'joined'", as we have 2 states, better use the direct link instead of indirect link.